### PR TITLE
Add Infinity87.5 PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Infinity87.5 PCB.
A TKL keyboard supports both the F12 and F13 footprints. 
Atmega32u4.
Thank you.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [ x I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/infinity875/config.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+#define LED_CAPS_LOCK_PIN D6
+#define LED_SCROLL_LOCK_PIN B4
+#define LED_PIN_ON_STATE 1
+
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_DI_PIN E2
+#define DRIVER_LED_TOTAL 42
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_WHEN_USB_SUSPENDED false // turn off effects when suspended
```
Comment (drashna) [inline keyboards/mechlovin/infinity875/config.h:56]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```
Or if you don't want it to sleep at all: 

```suggestion
// #define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/mechlovin/infinity875/info.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+    "keyboard_name": "Infinity87.5",
+    "url": "",
+    "maintainer": "Mechlovin' Studio",
+    "width": 18.25,
+    "height": 6.25,
```
Comment (drashna) [inline keyboards/mechlovin/infinity875/info.json:6]:
```suggestion
```

File: keyboards/mechlovin/infinity875/rules.mk [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/mechlovin/infinity875/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/mechlovin/infinity875/rules.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/mechlovin/infinity875/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/infinity875/rules.mk (hunk 3 above)
Comment (mechlovin) [reply keyboards/mechlovin/infinity875/rules.mk]: got it, thank you.

File: keyboards/mechlovin/infinity875/config.h [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+#define DISABLE_RGB_MATRIX_CYCLE_PINWHEEL	
+#define DISABLE_RGB_MATRIX_CYCLE_SPIRAL	
+#define DISABLE_RGB_MATRIX_DUAL_BEACON	
+#define DISABLE_RGB_MATRIX_RAINBOW_BEACON	
+#define DISABLE_RGB_MATRIX_RAINBOW_PINWHEELS	
+#define DISABLE_RGB_MATRIX_RAINDROPS	
+#define DISABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS	
+#define DISABLE_RGB_MATRIX_HUE_BREATHING	
+#define DISABLE_RGB_MATRIX_HUE_PENDULUM	
+#define DISABLE_RGB_MATRIX_HUE_WAVE
```
Comment (drashna) [inline keyboards/mechlovin/infinity875/config.h:85]: RGB Matrix requires explicit enabling of RGB Modes now. 

https://docs.qmk.fm/#/feature_rgb_matrix?id=rgb-matrix-effects

```suggestion
#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/infinity875/keymaps/default/config.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#ifdef RGB_MATRIX_ENABLE
+
+#define DISABLE_RGB_MATRIX_ALPHAS_MODS
```
Comment (drashna) [inline keyboards/mechlovin/infinity875/keymaps/default/config.h:22]: the disable defines are deprecated, and should be replaced with the enable defines, or undef's.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mechlovin/infinity875/rules.mk [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/mechlovin/infinity875/rules.mk:16]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/infinity875/rules.mk [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/mechlovin/infinity875/rules.mk:16]:
```suggestion
NKRO_ENABLE = no           # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by mechlovin** [comment]
@fauxpark a friendly ping in case you missed our earlier request. Our apologies if this comes at an inconvenient time.

File: keyboards/mechlovin/infinity875/rules.mk [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no           # Enable N-Key Rollover
```
Comment (fauxpark) [inline keyboards/mechlovin/infinity875/rules.mk:15]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by mechlovin** [comment]
@fauxpark Changes made according to request. We'd appreciate you taking the time to review and approve. Happy new year!

**Comment by mechlovin** [comment]
Hi @drashna & @fauxpark . We hope you're doing well and sorry for the ping again. Just wanted to ask if we could get some progress on this. Thank you.

File: keyboards/mechlovin/infinity875/infinity875.h [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E,   K0F,   K0G   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E,   K1F,   K1G   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E,   K2F,   K2G   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   KC_NO, KC_NO, KC_NO }, \
+    { K40,   K41,   K42,   K43,   K44,   K45,   K46,   K47,   K48,   K49,   K4A,   K4B,   K4C,   K4D,   KC_NO, K4F,   KC_NO }, \
+    { K50,   K51,   K52,   KC_NO, KC_NO, KC_NO, K56,   KC_NO, KC_NO, KC_NO, K5A,   K5B,   K5C,   K5D,   K5E,   K5F,   K5G   }, \
+}
+
+#define LAYOUT_88_ansi( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/infinity875.h:37]:
```suggestion
#define LAYOUT_tkl_f13_ansi_split_bs_rshift( \
```

File: keyboards/mechlovin/infinity875/infinity875.h [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E,   K0F,   K0G   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E,   K1F,   K1G   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E,   K2F,   K2G   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   KC_NO, KC_NO, KC_NO }, \
+    { K40, KC_NO,   K42,   K43,   K44,   K45,   K46,   K47,   K48,   K49,   K4A,   K4B,   K4C,   K4D,   KC_NO, K4F,   KC_NO }, \
+    { K50,   K51,   K52,   KC_NO, KC_NO, KC_NO, K56,   KC_NO, KC_NO, KC_NO, K5A,   K5B,   K5C,   K5D,   K5E,   K5F,   K5G   }, \
+}
+
+#define LAYOUT_88_iso( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/infinity875.h:53]:
```suggestion
#define LAYOUT_tkl_f13_iso_split_bs_rshift( \
```

File: keyboards/mechlovin/infinity875/infinity875.h [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E,   K0F,   K0G   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E,   K1F,   K1G   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E,   K2F,   K2G   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   KC_NO, KC_NO, KC_NO }, \
+    { K40,   K41,   K42,   K43,   K44,   K45,   K46,   K47,   K48,   K49,   K4A,   K4B,   K4C,   K4D,   KC_NO, K4F,   KC_NO }, \
+    { K50,   K51,   K52,   KC_NO, KC_NO, KC_NO, K56,   KC_NO, KC_NO, KC_NO, K5A,   K5B,   K5C,   K5D,   K5E,   K5F,   K5G   }, \
+}
+
+#define LAYOUT_87_ansi( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/infinity875.h:69]:
```suggestion
#define LAYOUT_tkl_ansi_split_bs_rshift( \
```

File: keyboards/mechlovin/infinity875/infinity875.h [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+) { \
+    { K00, KC_NO,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E,   K0F,   K0G   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E,   K1F,   K1G   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E,   K2F,   K2G   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   KC_NO, KC_NO, KC_NO }, \
+    { K40, KC_NO,   K42,   K43,   K44,   K45,   K46,   K47,   K48,   K49,   K4A,   K4B,   K4C,   K4D,   KC_NO, K4F,   KC_NO }, \
+    { K50,   K51,   K52,   KC_NO, KC_NO, KC_NO, K56,   KC_NO, KC_NO, KC_NO, K5A,   K5B,   K5C,   K5D,   K5E,   K5F,   K5G   }, \
+}
+
+#define LAYOUT_87_iso( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/infinity875.h:85]:
```suggestion
#define LAYOUT_tkl_iso_split_bs_rshift( \
```

File: keyboards/mechlovin/infinity875/info.json [hunk 14]
```diff
@@ -0,0 +1,23 @@
+{
+    "keyboard_name": "Infinity87.5",
+    "url": "",
+    "maintainer": "Mechlovin' Studio",
+    "layouts": {
+        "LAYOUT_87_ansi": {
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/info.json:6]:
```suggestion
        "LAYOUT_tkl_ansi_split_bs_rshift": {
```

File: keyboards/mechlovin/infinity875/info.json [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+        },
+        "LAYOUT_87_iso": {
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/info.json:9]:
```suggestion
        "LAYOUT_tkl_iso_split_bs_rshift": {
```

File: keyboards/mechlovin/infinity875/info.json [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+        },
+        "LAYOUT_88_ansi": {
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/info.json:12]:
```suggestion
        "LAYOUT_tkl_f13_ansi_split_bs_rshift": {
```

File: keyboards/mechlovin/infinity875/info.json [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+        },
+        "LAYOUT_88_iso": {
```
Comment (noroadsleft) [inline keyboards/mechlovin/infinity875/info.json:15]:
```suggestion
        "LAYOUT_tkl_f13_iso_split_bs_rshift": {
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**